File: 706B/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> sp_x(n);
    for (int i = 0; i < n; ++i) {
        cin >> sp_x[i];
    }
    sort(sp_x.begin(), sp_x.end());
    
    int q;
    cin >> q;
    for (int i = 0; i < q; ++i) {
        int m;
        cin >> m;
        auto it = upper_bound(sp_x.begin(), sp_x.end(), m);
        int can = distance(sp_x.begin(), it);
        cout << can << endl;
    }
    
    return 0;
}

File: 706B/main.py
import bisect

n = int(input())
sp_x = list(map(int, input().split()))
sp_x.sort()
q = int(input())
for _ in range(q):
    m = int(input())
    # Используем bisect_right для нахождения количества элементов <= m
    can = bisect.bisect_right(sp_x, m)
    print(can)

# bisect реализует бинарный поиск за нас, поэтому не нужно писать его вручную. Довольно удобно